addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-retention/nuxeo-retention-behavior.html

```diff
@@ -0,0 +1,149 @@
+<!--
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<script>
+  /* eslint-disable no-var, no-use-before-define */
+  var Nuxeo = Nuxeo || {};
+
+  (() => {
+    let fetcher;
+
+    function _fetchDocTypes() {
+      let typeFetcher = document.createElement('nuxeo-resource');
+      document.body.appendChild(typeFetcher);
+      typeFetcher.path = 'config/types';
+      return typeFetcher.get().then(async (res) => {
+        let docTypes;
+        let dateFields
+        if (res && res.doctypes) {
+          const schemas = new Set();
+          docTypes = [];
+          for (var type in res.doctypes) {
+            if (
+              res.doctypes[type].schemas.indexOf('file') !== -1 &&
+              res.doctypes[type].facets.indexOf('Folderish') === -1 &&
+              res.doctypes[type].facets.indexOf('HiddenInNavigation') === -1 &&
+              res.doctypes[type].facets.indexOf('SystemDocument') === -1
+            ) {
+              docTypes.push({
+                id: type,
+                text: type,
+              });
+              res.doctypes[type].schemas.forEach((schema) => {
+                schemas.add(schema);
+              });
+            }
+          }
+          dateFields = await _fetchDateFields(schemas);
+        }
+        return {docTypes, dateFields};
+      }).finally(() => {
+        document.body.removeChild(typeFetcher);
+        typeFetcher = null;
+      });
+    }
+
+    function _fetchDateFields(onlySchemas) {
+      if (onlySchemas.size > 0) {
+        let schemaFetcher = document.createElement('nuxeo-resource');
+        document.body.appendChild(schemaFetcher);
+        schemaFetcher.path = 'config/schemas';
+        return schemaFetcher.get().then((res) => {
+          const dateFields = [];
+          if (res) {
+            res.forEach((schema) => {
+              if (onlySchemas.has(schema.name)) {
+                for (var fieldName in schema.fields) {
+                  if (schema.fields[fieldName] === 'date') {
+                    dateFields.push(schema['@prefix'] + ':' + fieldName);
+                  }
+                }
+              }
+            });
+          }
+          return dateFields;
+        }).finally(() => {
+          document.body.removeChild(schemaFetcher);
+          schemaFetcher = null;
+        });
+      }
+    }
```

Comment:
Probably this would be an interesting candidate to extract since it is generic and useful in other contexts.